sqlserver/datadog_checks/sqlserver/metrics.py

```diff
@@ -410,19 +410,65 @@ class SqlDatabaseFileStats(BaseSqlServerMetric):
     QUERY_BASE = "select * from {table}".format(table=TABLE)
 
     DB_TYPE_MAP = {0: 'data', 1: 'transaction_log', 2: 'filestream', 3: 'unknown', 4: 'full_text'}
+    _DATABASES = []
+
+    def __init__(self, cfg_instance, base_name, report_function, column, logger):
+        super(SqlDatabaseFileStats, self).__init__(cfg_instance, base_name, report_function, column, logger)
+        self._DATABASES.append(self.instance)
 
     @classmethod
     def fetch_all_values(cls, cursor, counters_list, logger):
-        return cls._fetch_generic_values(cursor, None, logger)
+        # special case since this table is specific to databases, need to run query for each database instance
+        rows = []
+        columns = []
+
+        data = cursor.execute('select DB_NAME()').fetchall()
+        current_db = data[0][0]
+        logger.debug("%s: current db is %s", cls.__name__, current_db)
+
+        for db in cls._DATABASES:
+            # use statements need to be executed separate from select queries
+            ctx = 'use {}'.format(db)
+            logger.debug("%s: changing cursor context via use statement: %s", cls.__name__, ctx)
+            cursor.execute(ctx)
+            logger.debug("%s: fetch_all executing query: %s", cls.__name__, cls.QUERY_BASE)
+            cursor.execute(cls.QUERY_BASE)
+
+            data = cursor.fetchall()
+            query_columns = ['database'] + [i[0] for i in cursor.description]
+            if columns:
+                assert columns == query_columns
```

Comment:
I'm a bit confused about this line. Do you need to handle the case if `columns` != `query_columns` in case the `assert` fails?

Reply:
It's really just being overprotective that each of the queries have consistent column results in order to be merged together as one result.

If this assertion fails, then the check will abort .. which may be overkill, but also would be a sign of something very wrong happening